ipc/src/transport/blocking.rs

```diff
@@ -193,6 +193,10 @@ where
         self.transport.channel.set_write_timeout(timeout)
     }
 
+    pub fn is_closed(&self) -> bool {
+        self.transport.channel.probe_readable()
```

Comment:
The blocking transport is not supposed to be readable on the client side unless it's a response. So, outside of waiting for a response, it will never be readable unless the server side closed its socket.